lib/src/util/windows/user.cc

```diff
@@ -0,0 +1,58 @@
+#include <internal/util/windows/user.hpp>
+#include <internal/util/windows/process.hpp>
+#include <internal/util/windows/system_error.hpp>
+#include <internal/util/windows/windows.hpp>
+#include <facter/util/environment.hpp>
+#include <leatherman/logging/logging.hpp>
+
+using namespace std;
+
+namespace facter { namespace util { namespace windows { namespace user {
+
+    bool is_admin()
+    {
+        if (process::supports_elevated_security()) {
+            return process::has_elevated_security();
+        }
+
+        return check_token_membership();
+    }
+
+    bool check_token_membership()
+    {
+        DWORD sid_size = SECURITY_MAX_SID_SIZE;
+        unsigned char sid_buffer[SECURITY_MAX_SID_SIZE];
+        auto sid = static_cast<PSID>(&sid_buffer);
+        if (!CreateWellKnownSid(WinBuiltinAdministratorsSid, nullptr, sid, &sid_size)) {
+            LOG_DEBUG("Failed to create administrators SID: %1%", system_error());
+            return false;
+        }
+
+        if (!IsValidSid(sid)) {
+            LOG_DEBUG("Invalid SID");
+            return false;
+        }
+
+        BOOL is_member;
+        if (!CheckTokenMembership(nullptr, sid, &is_member)) {
+            LOG_DEBUG("Failed to check membership: %1%", system_error());
+            return false;
+        }
+
+        return is_member;
+    }
+
+    string home_dir()
+    {
+        string home, alt;
```

Comment:
Another note on the shell APIs (which were already in place at https://github.com/puppetlabs/facter/pull/907/files#diff-5264e60482bee614938ab1266bdf6ce4R39 as well)

* `SHGetFolderPath` has been deprecated by [SHGetKnownFolderPath](http://msdn.microsoft.com/en-us/library/bb762188%28v=vs.85%29.aspx)
* `SHGetFolderPath` / `SHGetKnownFolderPath` are actually not supported on `Server Core` as they're part of the shell APIs that have been removed.  I suspect the same will hold true for the newly announced `Nano Server`. See this [MSDN page](https://msdn.microsoft.com/en-us/library/hh846320\(v=vs.85\).aspx) for the list of APIs available in Server Core.  Note that `GetUserProfileDirectoryW` **is available** on that platform.

I know we don't explicitly make guarantees about running on `Server Core` (and certainly not `Nano Server`), but this is an area where I'd like to make sure we don't accidentally shoot ourselves in the foot.  In the future, I expect these Windows SKUs to become more important, so we don't want to leave Puppet out of the conversation on these platforms if we can help it.

/cc @joshcooper @ferventcoder  @kylog